scalalib/src/ScalaModule.scala

```diff
@@ -185,6 +187,11 @@ trait ScalaModule extends JavaModule { outer =>
       )
   }
 
+  override def docSources: Sources = T.sources(
+    if(isDotty(scalaVersion()) || isScala3Milestone(scalaVersion())) allSources()
+    else Seq(PathRef(compile().classes.path))
+  )
```

Comment:
I think, this is still not correct, e.g. for Scala 2. Best would be to explicitly guard the scala 3 case, as suggested previously.

```suggestion
  override def docSources: Sources = T.sources {
    // Scaladoc 3.0.0 is consuming tasty files
    if(isScala3(scalaVersion() && !isScala3Milestone(scalaVersion())) Seq((compile().classes))
    else allSources()
  }
```